src/utils/device.py

```
@@ -11,6 +11,7 @@ def using_gpu() -> bool:
 # by moving to some training framework + context objects once we have
 # benchmarking set up.
 def set_device(device: torch.device) -> None:
+    global _CHOSEN_DEVICE
     _CHOSEN_DEVICE = device
```

This was silently doing nothing. Great job, jder.

---

Lool! I was going to comment on that in the original PR, but I figured it was working. Great job, alxrms, too.